src/main/java/Storage.java

```diff
@@ -0,0 +1,200 @@
+     * @throws IOException Throws exception.
+     */
+    public Storage(String filePath) throws IOException {
+        // deals with loading tasks from the file and saving tasks in the dukeStorage.txt file.
+        this.filePath = filePath;
+        File file = new File(filePath);
+        file.createNewFile();
+    }
+
+    /**
+     * Load hard disk information into an ArrayList for TaskList class to use.
+     *
+     * @return Returns an ArrayList of Task, loaded from the Hard Disk.
+     * @throws FileNotFoundException Throws exception.
+     * @throws GrapieExceptions Throws exception.
+     */
+    public List<Task> load() throws FileNotFoundException, GrapieExceptions {
+        List<Task> storingList = new ArrayList<>();
+
+        File myObj = new File(filePath);
+        Scanner myReader = new Scanner(myObj);
+        while (myReader.hasNextLine()) {
+            String data = myReader.nextLine();
+
+            /*
+            T | 1 | read book
+            D | 0 | return book | June 6th
+            E | 0 | project meeting | Aug 6th 2-4pm
+            T | 1 | join sports club
+            */
+
+            String[] dataSplited = data.split("\\|");
+
+            //trim off empty spaces at front and back of string
+            for (int i = 0; i < dataSplited.length; i++) {
+                dataSplited[i] = dataSplited[i].trim();
+            }
+            //System.out.println(dataSplited.length);
+
+            if (dataSplited.length == 3) {
+                //is a todo
+                Task task = new Todo(dataSplited[2]);
+                if (dataSplited[1].equals("O")) {
+                    task.isDone = true;
+                }
+                storingList.add(task);
+
+            } else if (dataSplited.length == 4) {
+                //is a event or deadline
+                if (dataSplited[0].equals("E")) {
+                    //event
+                    Event task = new Event(dataSplited[2], dataSplited[3]);
+                    if (dataSplited[1].equals("O")) {
+                        task.isDone = true;
+                    }
+                    storingList.add(task);
+
+                } else {
+                    //deadline
+                    Deadline task = new Deadline(dataSplited[2], dataSplited[3]);
+                    if (dataSplited[1].equals("O")) {
+                        task.isDone = true;
+                    }
+                    storingList.add(task);
+                }
+            }
+        }
+
+        return storingList; //return the filled list
+    }
+
+    /**
+     * Convert task into correct format, and store in dukeStorage.txt file.
+     *
+     * @param task The task to be converted into the new format.
+     * @param type Todo, Event or Deadline.
+     * @param time The date and time for the Tasks.
+     * @throws IOException Throws exception.
+     */
+    public void convertToHardDiskFormatAndStore(Task task, String type, String time) throws IOException {
```

Comment:
Maybe save might be a simpler name for this method? You could always put this name in the description.